### PR TITLE
fix(learn): Headless Browser Tests have incorrect values

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

The headless browser tests are testing for the incorrect assertions, causing the tests to erroneously fail. This PR fixes the tests and aligns the instructions with the correct information.

See: https://forum.freecodecamp.org/t/chai-lesson-tests-pass-in-console-but-not-on-fcc/432605

### CODE REVIEW

**Comment by Sky020** [comment]
@nhcarrigan I am very confused what is going on... If I look at my fork's master: https://github.com/Sky020/freeCodeCamp/blob/master/curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/simulate-actions-using-a-headless-browser.md


These suggested changes are already in production?

**Comment by Sky020** [comment]
The issue appears to be this file: https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/run-functional-tests-using-a-headless-browser.md

The tests do not match the instructions.

**Comment by nhcarrigan** [comment]
So, rather than the current change, I need to move the tests to the correct challenge files? I can do that, certainly - just want to make sure I am on the right page. 🙂

**Comment by Sky020** [comment]
@nhcarrigan I have messaged you on `chat.freeCodeCamp` with the idea.

**Comment by nhcarrigan** [comment]
Okay,

@Sky020 I compared the boilerplate refactor to the previous codebase. We spoke on chat and you mentioned a `III` file, but it looks like that was the "Simulate Actions Using a Headless Browser" file instead (which was newly added in the mocha/chai refactor. To maintain a logical flow of information, I switched the names around so the introductory text (before the camper writes any tests) is under that header, and the two files that involve writing actual test assertions are under the "Run Functional Tests on a Headless Browser" header.

It looks like the intent here was to keep the setup text in its own file (I assume this is information that was migrated out of the boilerplate?). I removed the tests from that file as they were not correlated to that challenge's instructions. However, the boilerplate doesn't include an `_api` endpoint to read the `suiteSetup` code so I am not sure how we want to go about testing this one. For now, I went ahead and left an "All tests should be passing" call that looks at a test the camper previously wrote (so should pass regardless). Any thoughts on how to approach this part?

/cc @RandellDawson

File: curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/simulate-actions-using-a-headless-browser.md [hunk 1]
```diff
@@ -8,93 +8,53 @@ challengeType: 2
...
 
-You should assert that the element 'span#dates' exist and its count is 1.
+# --hints--
+
+All tests should pass.
 
 ```js
 (getUserInput) =>
-  $.get(getUserInput('url') + '/_api/get-tests?type=functional&n=6').then(
+  $.get(getUserInput('url') + '/_api/get-tests?type=functional&n=0').then(
```
Comment (Sky020) [inline curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/simulate-actions-using-a-headless-browser.md:55]: This just checks if the first functional test has passed. Do we not want it to check if all previous 4 are passing, instead?

**Review by ShaunSHamilton — COMMENTED** [review]
@nhcarrigan , This is much better, I think. The naming probably is not the best, but this way around makes more sense.

Other than the one comment to potentially change the test to look for the state of the previous 4, I would merge this as is, and sort the tests out in another PR which involves the boilerplate as well.

File: curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/simulate-actions-using-a-headless-browser.md (hunk 1 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/simulate-actions-using-a-headless-browser.md]: Sure - I've changed that 🙂

**Review by naomi-lgbt — COMMENTED** [review]
https://repl.it/@NicholasCarriga/boilerplate-mochachai - I've got a repl.it for testing these changes against, built from the new boilerplate. 

Ran into a small tweak needed on the boilerplate to get the tests to return properly for the `get-tests` endpoint so will ship a PR there shortly.

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Comment by RandellDawson** [comment]
So what was the final reason this PR is needed?  What did we miss in our reviews of the PRs that were recently merged?

**Comment by nhcarrigan** [comment]
We missed a mis-alignment of the test cases vs. the challenge instructions for these two lessons:
https://www.freecodecamp.org/learn/quality-assurance/quality-assurance-and-testing-with-chai/run-functional-tests-using-a-headless-browser
https://www.freecodecamp.org/learn/quality-assurance/quality-assurance-and-testing-with-chai/run-functional-tests-using-a-headless-browser-ii

Which occurred I believe with the addition of this lesson:
https://www.freecodecamp.org/learn/quality-assurance/quality-assurance-and-testing-with-chai/simulate-actions-using-a-headless-browser

**Comment by RandellDawson** [comment]
So it was not just a case that the meta.json file was in the wrong order?

**Comment by nhcarrigan** [comment]
Correct - I did reorder the challenges to match the logical flow of the instructions, but that was not the source of the error.

**Comment by Sky020** [comment]
@RandellDawson  I am pinging and bumping this up as it would be better to get fixed sooner, rather than later.

**Comment by RandellDawson** [comment]
@Sky020 We are currently in a code freeze for curriculum PRs.

**Comment by Sky020** [comment]
@RandellDawson Ah. I thought that was finished.

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Comment by arnav-t** [comment]
Hi, when will this be deployed in production? The tests are still failing as of 23rd December 2020. I forked the boilerplate about 2-3 days ago.

**Comment by ShaunSHamilton** [comment]
Hey @arnav-t ,You can keep track by looking for the PR in: https://github.com/freeCodeCamp/freeCodeCamp/commits/production-current

There is no hard-fast rule for when `master` changes are pushed to `production`, but I usually expect _10 working days_.

**Comment by arnav-t** [comment]
> Hey @arnav-t ,You can keep track by looking for the PR in: https://github.com/freeCodeCamp/freeCodeCamp/commits/production-current
> 
> There is no hard-fast rule for when `master` changes are pushed to `production`, but I usually expect _10 working days_.

Thanks for the information. Looking forward to completing the certification! 😄

**Comment by andreibexa** [comment]
I passed both tests after I removed second suite, so only top suite should remain "Integration tests with chai-http"

// `suite("e2e Testing with Zombie.js", function () {`

**Comment by ShaunSHamilton** [comment]
To whoever comes across this:

Seeing as both the `/learn` platform tests, as well as the boilerplate have been updated, in order to pass these challenges, it is necessary to re-clone the boilerplate.

_Do I need to start from scratch?_
- Unless you are comfortable making the necessary changes to your existing boilerplate to match the test code of the current boilerplate, it **is** advisable to start from scratch.

If you have any further questions, please make use of the [forum](https://forum.freecodecamo.org/)

**Comment by gianlucas90** [comment]
Hello, I have re-cloned the repl.it repository but I have the same issue. 
https://repl.it/@GianlucaZin/boilerplate-mochachai-2

**Comment by ShaunSHamilton** [comment]
@gianlucas90 This line in the updated boilerplate does not match your code: https://github.com/freeCodeCamp/boilerplate-mochachai/blob/9ba0fb710a1fa39b8d33f154e8004e1c96478f9b/tests/2_functional-tests.js#L56

In general, I recommend you open a topic on the [forum](https://forum.freecodecamp.org/) for help with challenges/code.

**Comment by gianlucas90** [comment]
@ShaunSHamilton thank you very much!

**Comment by nilueps** [comment]
I've also cloned the latest boilerplate and am still encountering this issue.
https://repl.it/@nilueps/boilerplate-mochachai#tests/2_functional-tests.js

**Comment by ShaunSHamilton** [comment]
@nilueps I recommend you open a topic on the [forum](https://forum.freecodecamp.org/) with information like:
- Link to project code
- Error you are encountering
- Link to lesson you are stuck on
- Information you can see in
  - Browser console when you submit
  - Repl.it console

**Comment by jawwad** [comment]
The latest change did fix it for me. I didn't reclone but just made the text change that was made in https://github.com/freeCodeCamp/boilerplate-mochachai/commit/9ba0fb710a1fa39b8d33f154e8004e1c96478f9b manually and it worked for me.

**Pull request merged**